ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionComponent.java

```diff
@@ -235,6 +235,10 @@ public AccessTimeoutDetails getAccessTimeout(Method method) {
     }
 
     public SessionID createSession() {
+        // check to see that StatefulSessionComponent.stop() has not been called concurrent with the request; this can happen on undeploy (see WFLY-5178)
+        if (getCache() == null)
```

Comment:
This is just a racey hack. The cache can still become null after this check resulting in the same exception. There is no way of preventing these exceptions when the container is shutting down, graceful shutdown is the only real solution.